src/common.ts

```diff
@@ -1,5 +1,5 @@
 import { Rule } from "eslint";
-import { TSESTree } from "@typescript-eslint/types";
+import { type TSESTree } from "@typescript-eslint/types/dist";
```

Comment:
I'd prefer we leave this PR only with the fix, and another for the package changes. I need to analyze if there's a better way of doing the types, since it looks strange to me that some types had to be imported from a dir like this

Reply:
Sure, will prepare the smaller pr in few hours!